### DISCUSSION TITLE
Add encryption status ?

### DISCUSSION THREAD

**Discussion by psafr1** [waa/bacula#13] [Ideas]
Hello,

first of all, many thanks for your work.

Do you think it is possible to add a column with the job encryption status ?
I think that it's the last column (encrypted) in the job table.

Regards

**Comment by waa** [comment]
Hello!

I just saw this request now. No idea how I missed it when you sent it 3 weeks ago!

You are welcome! Glad there is someone using my script(s). :)

I think this is a good idea.

Yes, when I have a chance, I will work on adding this.

Thank you for the feature request!


... Looking at this now. I already see some issues. bugs?

I am seeing in my catalog that for Copied/Migrated jobs written to SD encrypted disk volumes, the `encrypted` field is 0, and the job summary also shows:
```
Encryption:             no
```

I just ran a plain Backup job to SD encrypted disk volumes,  the `encrypted` field is also 0.

Additionally, the job summary for this same backup job to encrypted disk volumes shows:
```
Encryption:             no
```

My guess is two things here. 

1. The encrypted field in the catalog, and in the Job summary only show 1/yes respectively when Job encryption is done in the legacy way - by the file daemon.
2. I will need to open a bug report/feature request with the developers to flag the new "Storage Encrypted" jobs so that they set the DB encrypted field to 1, and show `Encryption:             yes` in job summaries.

Or better, `Encryption:  (SD|FD|BOTH)` depending on the way the data was encrypted.

Then the DB encrypted field would need to be one of something like:
```
- 0 No encryption
- 1 FD encrypted
- 2 SD encrypted
- 3 FD and SD encrypted
```
Let me see what I can do internally with the developers and I will come back to you about this request. 

I will probably start some work on it though. First, to test my theory that this is currently only when FD encryption is used, and then adapt my script to that, and then make modifications if the developers agree to my above ideas. :)

Thank you again for using my script(s), and for the feature request.

OH, can you tell me how you are doing encryption and if you are seeing any different results in the catalog or in the Job summaries than I have described above?


Best regards,
Bill

**Comment by psafr1** [comment]
Hello Bill,

I'm currently using FD encryption using certificate, cross signed by DIR public key (DIR private key is not stored on the DIR server for security reason).
I prefer this way to get end to end encryption and privacy.

I got this information from job log :
 ` Encryption:             yes`
And in database (using postgres) :
in job table, field "encrypted" (type integer") is marked as 1  
I'm not using it, but in media table, there's a "volencrypted" field (type smallint).

PS: i'm using bacula-dir 15.0.2. Theses fields seems to havve been added only in 15.X (https://www.bacula.org/15.0.x-manuals/en/main/New_Features_in_15_0_0.html)

Regards,
Philippe.

**Comment by waa** [comment]
Hello Philippe,

Yes, re: "I'm currently using FD encryption" , that is exactly what I expected to hear. :)

Also, "Storage Daemon Encryption" was added in Bacula Enterprise v16.x.y, so it was also included in Community v15.x.y - the version that tracks Enterprise 16... So that makes sense. 

So, I can start to work on adding support for an "Encrypted" column while I wait on feedback from the developers about flagging Storage Daemon Encrypted jobs as "Encrypted: yes" in the Job table and Job Summary. 

And also, yes. If you do `ll volume=xxxx` there is a new `encrypted: 1`  field for my Storage Daemon Encrypted volumes, so most of the work is already there. :)


Best regards,
Bill

**Reply by psafr1** [reply]
Hello Bill,

In my case, on `ll volume=xxxx`,  this new volencrypted is 0 :`volencrypted: 0`.
That's OK because I'm not using Volume Encryption, only FD.

Regards.
Philippe.

**Comment by waa** [comment]
Hello Philippe!

I have two kinds of good news for you!

First, I have pretty much completed this new "encrypted column" feature. (see screenshot below)

Second, my colleagues on the Bacula Systems R&D team confirmed that the new "Storage Daemon Encryption" feature not updating the DB and the Job Summary was missed when the feature was created.

They liked my ideas and we now will soon have:
- 0 - No encryption
- 1 - FD encryption
- 2 - SD (volume) encryption
- 3- Both

These of course will be reflected and updated in the Job table in the catalog in the next release. It will not affect you since I suspect you are not using SD encryption yet. So this should already work fine for you.

I am just cleaning things up in my script and then I will merge this new "encryption" branch and push my changes.

To test, you will need to add 'encrypted' to your `cols2show` variable, then have a look at and set the `enc_cell_type`, `enc_emoji`, and `un_enc_emoji` variables. Defaults are already OK, but you may like different emojis, or none at all.

Please open an `issue` here in Gihub if you have a :lady_beetle: to report on this, or any feature. 

Thank you again for the excellent feature request.

![image](https://github.com/waa/baculabackupreport/assets/108133/0bc5a8cf-a103-4036-a937-bbcd7c264094)

**Comment by psafr1** [comment]
Hello Bill,

Again, many thanks for all your work !

![image](https://github.com/waa/baculabackupreport/assets/104484073/3e1487a5-4c00-433f-acb0-ced37c868e20)

This is perfect for me.
Have a nice week!

Regards.
Philippe

**Comment by waa** [comment]
Glad it is working for you!

I'll go ahead and close this discussion now. :)

Best regards,
Bill
